src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionIT.java

```diff
@@ -1,62 +1,231 @@
+import com.snowflake.kafka.connector.dlq.InMemoryKafkaRecordErrorReporter;
+import com.snowflake.kafka.connector.internal.DescribeTableRow;
+import com.snowflake.kafka.connector.internal.SnowflakeSinkService;
+import com.snowflake.kafka.connector.internal.SnowflakeSinkServiceFactory;
+import com.snowflake.kafka.connector.internal.TestUtils;
+import com.snowflake.kafka.connector.internal.streaming.InMemorySinkTaskContext;
+import com.snowflake.kafka.connector.internal.streaming.IngestionMethodConfig;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
 import java.util.HashMap;
-import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
-import java.util.Properties;
-import net.snowflake.ingest.streaming.OpenChannelRequest;
-import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
-import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
-import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
-public class IcebergIngestionIT {
-  private static final String tableName = "tbl1";
+public class IcebergIngestionIT extends BaseIcebergIT {
+
+  private String tableName;
   private static final int PARTITION = 0;
-  private static final String topic = tableName;
-  private static final String testChannelName =
-      SnowflakeSinkServiceV2.partitionChannelKey(topic, PARTITION);
+  private String topic;
+  private TopicPartition topicPartition;
 
-  @Test
-  @Disabled
-  void shouldInsertValue() throws Exception {
-    ObjectMapper mapper = new ObjectMapper();
-    Properties props = new Properties();
-    Iterator<Map.Entry<String, JsonNode>> propIt =
-        mapper.readTree(new String(Files.readAllBytes(Paths.get(PROFILE_PATH)))).fields();
-    while (propIt.hasNext()) {
-      Map.Entry<String, JsonNode> prop = propIt.next();
-      props.put(prop.getKey(), prop.getValue().asText());
-    }
-    SnowflakeStreamingIngestClient client =
-        SnowflakeStreamingIngestClientFactory.builder("name")
-            .setProperties(props)
-            // .setIsIceberg(true)
-            .build();
+  private SnowflakeSinkService service;
+
+  private final String simpleRecordJson = "{\"simple\": \"extra field\"}";
+
+  private static final String primitiveJson =
+      "{\n"
+          + "  \"id_int8\": 0,\n"
```

Comment:
Maybe we can skip `\n`?

Reply:
good idea, it does not matter to jackson anyways